src/registry/domain/repository.js

```diff
@@ -183,7 +189,10 @@ module.exports = function(conf){
         return callback(null, 'ok');
       }
 
-      componentsCache.load(callback);
+      componentsCache.load((err, componentsList) => {
+        if(err){ return callback(err); }
+        componentsDetails.refresh(componentsList, (err) => callback(err, componentsList));
+      });
```

Comment:
This is what happens on registry start. In this scenario, the `componentsDetails.refresh` main goal is to verify integrity or to build the file from scratch (only the first time).